Populate '[event][sequence]' instead of 'clock' if ECS is enabled (#18)
logstash-plugins/logstash-input-heartbeat: docs/index.asciidoc

The general intention of this is to test the performance and
availability of Logstash.

[id="plugins-{type}s-{plugin}-ecs"]
==== Elastic Common Schema (ECS)

> This plugin could provide a field, originally named `clock`, to track `epoch` or `sequence` incremental numbers. When
> <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled that value is now present in the
> event's `[event][sequence]` subfield.
> When <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled the use of `message` as
> selector of sequence type is not available and only <<plugins-{type}s-{plugin}-sequence>> is considered. In this
> case if `message` contains sequence selector strings it's ignored.

Reviewer: ```suggestion
This plugin provides a field, originally named `clock`, to track `epoch` or `sequence` incremental numbers. When
<<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled, that value is now present in the
event's `[event][sequence]` subfield.

When <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled the use of `message` as
selector of sequence type is not available and only <<plugins-{type}s-{plugin}-sequence>> is considered. In this
case if `message` contains sequence selector strings, it is ignored.
```

The general intention of this is to test the performance and
availability of Logstash.

[id="plugins-{type}s-{plugin}-ecs"]
==== Elastic Common Schema (ECS)

This plugin could provide a field, originally named `clock`, to track `epoch` or `sequence` incremental numbers. When
<<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled that value is now present in the
event's `[event][sequence]` subfield.

When <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled the use of `message` as
selector of sequence type is not available and only <<plugins-{type}s-{plugin}-sequence>> is considered. In this
case if `message` contains sequence selector strings it's ignored.

The existing `host` field is moved to `[host][name]` when ECS is enabled.

> .Metadata Location by `ecs_compatibility` value
> [cols="<l,<l,e,<e"]
> |============================================================================================================
> |`disabled`    |`v1`              |Availability                       |Description

Reviewer: ```suggestion
[cols="<l,<l,e,<e"]
|============================================================================================================
|`disabled`    |`v1`              |Availability                       |Description

```

Reviewer: Removes table title. Adds return after table headings to make them format as bold.

How many times to iterate.
This is typically used only for testing purposes.

[id="plugins-{type}s-{plugin}-ecs_compatibility"]
===== `ecs_compatibility`

* Value type is <<string,string>>
* Supported values are:
** `disabled`: `clock` counter field added at root level
** `v1`: ECS compliant `[event][sequence]` counter field added to the event
* Default value depends on which version of Logstash is running:
** When Logstash provides a `pipeline.ecs_compatibility` setting, its value is used as the default
** Otherwise, the default value is `disabled`.

> Refer to <<plugins-{type}s-{plugin}-ecs,ECS mapping>> for detailed information.

Reviewer: ```suggestion
Controls this plugin's compatibility with the {ecs-ref}[Elastic Common Schema (ECS)]. 
Refer to <<plugins-{type}s-{plugin}-ecs>> in this topic for detailed information.
```

Reviewer: The link probably shouldn't say "ECS mapping" since the ECS topic in this plugin doesn't have that title.

Otherwise, this value will be used verbatim as the event message. It
will output this value into a field called `message`

NOTE: Usage of `epoch`  and `sequence` in `message` setting is deprecated.

> Consider to use <<plugins-{type}s-{plugin}-sequence>> settings, which takes precedence

Reviewer: ```suggestion
Consider using <<plugins-{type}s-{plugin}-sequence>> settings, which takes precedence
```

> over the usage of `message setting as selector.

Reviewer: Looks like the ending backtick(`) is missing. Should it go after "message"?

Reviewer: ```suggestion
over the usage of `message` setting as selector.
```

> NOTE: If <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled
> and `message` contains `epoch` or `sequence` it's ignored and it's not present as a field in
> the generated event.

Reviewer: ```suggestion
NOTE: If <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>> is enabled
and `message` contains `epoch` or `sequence`, it is ignored and is not present as a field in
the generated event.
```

[id="plugins-{type}s-{plugin}-sequence"]
===== `sequence`

* Value can be any of: `none`, `epoch`, `sequence`

> * Default value is `"none""`

Reviewer: ```suggestion
* Default value is `"none"`
```

> If you set this to `none` then no sequence field is added.

Reviewer: ```suggestion
If you set this value  to `none`, then no sequence field is added.
```

> If you set this to `epoch` then this plugin will use the current

Reviewer: ```suggestion
If you set this value to `epoch`, then this plugin will use the current
```

timestamp in unix timestamp (which is by definition, UTC).  It will
output this value into a field called `clock` if <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>>
is disabled else the field name is `[event][sequence]`.

> If you set this to `sequence` then this plugin will send a sequence of

Reviewer: ```suggestion
If you set this value to `sequence`, then this plugin will send a sequence of
```

numbers beginning at 0 and incrementing each interval.  It will

> output this value into a field called `clock` if <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>>
> is disabled else the field name is `[event][sequence]`

Reviewer: ```suggestion
output this value into a field called `clock` if <<plugins-{type}s-{plugin}-ecs_compatibility,ECS compatibility mode>>
is disabled. Otherwise, the field name is `[event][sequence]`
```

> If `sequence` is defined it takes precedence over `message` configuration; in case `message` doesn't
> have `epoch` or `sequence` values, it will be present also in the generated event together with `clock`
> field.

Reviewer: ```suggestion
If `sequence` is defined, it takes precedence over `message` configuration.
If `message` doesn't have `epoch` or `sequence` values, it will be present in the generated event together with `clock`
field.
```

Author: @karenzone thanks for all your suggestions, I've integrated into the documentation file, plus I've added a summary table for the ECS remappings done by this plugin
